fleetspeak/src/server/sqlite/messagestore.go

```diff
@@ -402,10 +402,59 @@ func genPlaceholders(num int) string {
 	return strings.Join(es, ", ")
 }
 
-func (d *Datastore) DeletePendingMessages(ctx context.Context, ids []common.ClientID) error {
-	return d.runInTx(func(tx *sql.Tx) error {
+func (d *Datastore) getPendingMessageRawIds(ctx context.Context, tx *sql.Tx, ids []common.ClientID, offset uint64, limit uint64) ([]string, error) {
```

Comment:
Same question as for the MySQL implementation: I wonder if instead of doing the query in 2 steps, we could do a single SELECT over 2 tables?

Reply:
See comment above.